docs/naming_conventions.md

```diff
@@ -0,0 +1,52 @@
+The platform impose several constraints with regards to the naming of the persistence elements (files, databases, tables and collections). Such constraints, and the way the different storages solve them, must be had into account when designing the data model of the solution running on top of the platform.
```

Comment:
persistence element -> persisentence container? ("element" may refer to the container or to the thing being persisted and I understand that the naming conventions defined in this document refers to the former).